src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -46,9 +56,38 @@ public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> call
     {
         ArgumentNullException.ThrowIfNull(callback);
 
-        _registeredCallbacks.Add(callback);
+        var serializationMode = _serializationModeHandler.GetCallbackTargetSerializationMode(callback.Target);
+
+        return RegisterOnPersisting(callback, serializationMode);
+    }
+
+    /// <summary>
+    /// Register a callback to persist the component state when the application is about to be paused.
+    /// Registered callbacks can use this opportunity to persist their state so that it can be retrieved when the application resumes.
+    /// </summary>
+    /// <param name="callback">The callback to invoke when the application is being paused.</param>
+    /// <param name="serializationMode">The <see cref="PersistedStateSerializationMode"/> to register the callback.</param>
+    /// <returns>A subscription that can be used to unregister the callback when disposed.</returns>
+    public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> callback, PersistedStateSerializationMode serializationMode)
+    {
+        ArgumentNullException.ThrowIfNull(callback);
+
+        if (_serializationModeHandler.GlobalSerializationMode != PersistedStateSerializationMode.Infer &&
+            serializationMode != _serializationModeHandler.GlobalSerializationMode)
```

Comment:
Can you enumerate the possible combinations for the different use cases? It's not clear how this works E2E

Reply:
This means "Unless the GlobalSerializationMode is not infer you cannot persist state for serializationMode different that the GlobalSerializationMode".
I'll provide a clarifying comment, since it raised a question.